ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -245,14 +247,17 @@ public Response handleKsqlStatements(
           request,
           distributedCmdResponseTimeout);
 
+      final KsqlRequestConfig requestConfig =
```

Comment:
I am not sure about initializing the config with every request. The config has the default value, you should not override it. Instead, you can check if there is the property specified in the request and use that value, else use the default value.

```     
if (request.getRequestProperties().containsKey(key)) {
        return (Boolean) request.getRequestProperties().get(key);
}
else {
    return KsqlRequestConfig. KSQL_REQUEST_INTERNAL_REQUEST;
}
```

Reply:
@vpapavas I'm not overriding the default, if the config isn't present and you create a new KsqlRequestConfig, it should automatically set to what the default value is. I wrote a new test file to illustrate the behavior. https://github.com/confluentinc/ksql/blob/b66866b7a3f76547b217579935bc00240854d530/ksqldb-common/src/test/java/io/confluent/ksql/util/KsqlRestConfigTest.java